### ISSUE TITLE
Would you please help me troubleshoot why motion alert notifications are not working?

### ISSUE THREAD

**Issue by jcg3** [hjdhjd/homebridge-unifi-protect#454]
**Describe The Problem:**
Motion alerts from my doorbell camera are NOT triggering notifications in the Home app.

Doorbell ring alerts ARE showing up in the Home app with a push notification.

I do have motion alerts turned on in the Home settings for my iPhone. I turned it on for my iMac too as part of troubleshooting - same behavior on the Mac.

I have alerts turned on in the Protect app on my phone and the motion alerts are properly showing up as notifications, doorbell rings too.

A while back, the motion alerts were correctly triggering notifications in the Home app, but they stopped working at some point recently - for both my wife and I. Unfortunately, I'm not sure when exactly this happened.

Per the Troubleshooting page, I confirmed the Cloud Key and my computer running Homebridge had the same system time:

> $ ssh unifi.local date ; date
> Thu Jul  1 14:34:58 EDT 2021
> Thu Jul  1 14:34:58 EDT 2021

Looking for any suggestions to further troubleshoot what might be causing the problem. Can I turn something on for extra logging? Are there logs on the Cloud Key that I can look into?

I'm suspicious that the problem is in my UniFi Protect somewhere... some unrelated issues that I'm having:
- Protect is not sending me emails when motion is detected, even though I have those alerts set up. I used to get emails, but then they were sporadic for a while, and now I'm not getting any. It's most likely gmail blocking the emails (they've been overly aggressive on blocking emails for me from other automated senders too), but could be UniFi's problem.
- When I open the Protect app on my phone, it shows a status of "Remotely Connected" even though I'm on the same WiFi. Quick googling suggests this is not uncommon, but I figured it was worth mentioning.

Something suspicious regarding HomeKit - when I went into the Home app on my iPhone and looked at my Home Settings for the doorbell camera, motion alerts were turned off. I don't remember turning those off - maybe an Apple HomeKit bug? Maybe from me logging out of iCloud and logging back in? I turned the alerts setting back on, went outside and unfortunately had the same behavior - doorbell alert, but no motion alert in Home. Maybe HomeKit is buggy with the setting really being saved properly?

**To Reproduce:**
Walk in front of the camera, see the alert from the Protect app but not in the Home app.

**Logs:**
I don't think these are related, but this is around the time I walked outside and triggered a motion alert, and I also pushed the doorbell to confirm behavior described above.

```
[7/1/2021, 2:21:23 PM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/1/2021, 2:26:18 PM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/1/2021, 2:27:19 PM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
```

**Homebridge Configuration:**

```json
        {
            "controllers": [
                {
                    "address": "192.168.1.2",
                    "password": "SECRET",
                    "username": "SECRET"
                }
            ],
            "platform": "UniFi Protect"
        },
```

**Screenshots:**
![Home Setting](https://user-images.githubusercontent.com/140759/124174136-ff0ef380-da79-11eb-9e77-75c22925f330.jpeg)
![alerts2](https://user-images.githubusercontent.com/140759/124174138-ff0ef380-da79-11eb-8d89-90d6276c32eb.jpeg)

**Environment:**

* **Homebridge Version:** 1.3.4
* **Node Version:** 14.15.3
* **Homebridge-UniFi-Protect Plugin Version:** 4.4.1
* **UniFi Protect Version:** 1.18.1
* **UniFi Protect Controller Type:** UCKgen2+
* **UniFi Protect Controller Firmware Version:** 2.1.11
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** iMac running homebridge on macOS 11.4, iPhone 11 Pro Max running the Home app on iOS 14.6
* **Operating System and OS Version:** macOS 11.4

Not sure if it matters, but the camera in question is a G4 Doorbell, firmware version 4.38.1.

**Comment by hjdhjd** [comment]
Seems like you got this one solved on your own. :smile: If you're still having issues, please let me know. Thanks!

**Comment by jcg3** [comment]
Yes, I'm still having issues. The motion notifications in the Home app are not working.

Is there a way to turn on additional logging, or logs I can check on the cloud key that might help point to where things are breaking down?

Thanks

**Comment by hjdhjd** [comment]
RTFM. :smile: 

https://github.com/hjdhjd/homebridge-unifi-protect/blob/master/docs/FeatureOptions.md#logging

**Comment by jcg3** [comment]
Now we're getting somewhere, thanks for pointing me to the logging options. I enabled logging for both motion and doorbell in my config, restarted homebridge.

I don't see anything in the logs when I trigger a motion alert. I am still getting the push notification on my phone from the Protect app, so it is detecting motion, just dropping the ball somewhere...

Here are the logs I get when I press the doorbell button:

```
[7/3/2021, 11:00:52 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Doorbell ring detected.
[7/3/2021, 11:01:03 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Streaming request from 192.0.2.1: 320x240@15fps, 68 kbps. Using RTSP stream profile: 480x360@15fps (Low), 200 kbps.
[7/3/2021, 11:01:05 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Stopped video streaming session.
```

The only odd thing I see is the ip-address 192.0.2.1 -- that's not a subnet I use anywhere on my network. If that's expected behavior, then we can ignore it, I just figured it was worth pointing out.

So, it seems that Homebridge and the plugin don't see the motion alert at all... am I reading that correctly? Suggestions for next steps of troubleshooting?

**Comment by jcg3** [comment]
This may or may not be related, but I added a few more options to my config, in addition to the logging... even though I turned on smart detect object sensors, I'm only seeing one sensor at a time in the Home app. One time it showed a Person sensor, I restarted Homebridge and the next time it showed a Vehicle sensor... 

```
            "platform": "UniFi Protect",
            "controllers": [
                {
                    "address": "192.168.177.2",
                    "password": "secret",
                    "username": "secret"
                }
            ],
            "options": [
                "Enable.Log.Doorbell",
                "Enable.Log.Motion",
                "Enable.Motion.SmartDetect.NvrEvents",
                "Enable.Motion.SmartDetect.ObjectSensors",
                "Enable.Motion.Switch",
                "Enable.Motion.Trigger",
                "Disable.Doorbell.Messages"
            ]
```

Here's what shows in the logs from a restart, as well as me walking outside and pushing the doorbell.

```
[7/3/2021, 11:39:08 AM] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[7/3/2021, 11:39:08 AM] [UniFi Protect] Initializing UniFi Protect platform...
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the Protect controller API (address: 192.168.177.2 mac: 7483C273D68B).
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Discovered camera: Front Door [UVC G4 Doorbell] (address: 192.168.177.10 mac: E063DAEF1A7D).
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Controller refresh interval set to 10 seconds.
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Enabling motion sensor switch.
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Enabling motion sensor automation trigger.
[7/3/2021, 11:39:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Smart motion contact sensors enabled: person, vehicle.
[7/3/2021, 11:40:41 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/3/2021, 11:41:52 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/3/2021, 11:42:12 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/3/2021, 11:42:43 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS]: Connected to the UniFi realtime update events API.
[7/3/2021, 11:43:05 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Doorbell ring detected.
[7/3/2021, 11:43:09 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Streaming request from 192.168.177.89: 320x240@15fps, 68 kbps. Using RTSP stream profile: 480x360@15fps (Low), 200 kbps.
[7/3/2021, 11:43:14 AM] [UniFi Protect] Slate Blue [UCK-G2-PLUS] Front Door [UVC G4 Doorbell]: Stopped video streaming session.
```

So the plugin thinks that both person and vehicle should be there, but it's not showing up in the Home app. Curious to say the least. And no motion, person, nor vehicle alerts were triggered despite me going outside and pushing the doorbell button again.

I assume it is normal for the plugin to re-connect to the realtime update events API several times in a row...

**Comment by hjdhjd** [comment]
You've got too much going on to meaningfully troubleshoot.

Simplify your setup to the basics. Get the most basic configuration working **first**, and then add increasing complexity / customization until you run into a problem.

As to what IP addresses show up in the logs - if it's not an IP you recognize, you may want to investigate it. I can't help you troubleshoot your network setup I'm afraid. :smile: When a request is made to stream video in HomeKit, the plugin responds by streaming that video to that IP address. You're on your own from there.

Finally - notifications in the Home app are out of scope for any support I can provide, I'm afraid. All I can ultimately tell you is whether or not the notification / alert was passed on to HomeKit. There's no meaningful visibility to the plugin (or really anyone I'm afraid) to let you know if the Home app itself is seeing it or not and generates a notification to the end user. The notification's being generated, and sent on it's way from the plugin. Beyond that...you're going to need to figure out what's going on within the Home app and it's notification settings.

For what it's worth, my general sense is that you've tried to turn on a lot of things here. I'd strongly suggest the best practice of starting simple. Make a single change. Test. Verify. Then make another change. It's slower, certainly, but it's tried-and-true.

**Comment by jcg3** [comment]
I'm just trying to share all the weird stuff that I am seeing in case any of them are relevant or if they provide clues. If not, let's ignore all those details.

Let me focus on the simple case. I have motion alerts turned on. I have enabled the motion logs (Enable.Log.Motion).

I walk in front of the camera:
- I get a motion alert in the UniFi Protect app
- **There is nothing logged to the homebridge log**
- I do not get a notification in the Home app.

To contrast with the above, the doorbell alerts are working as expected. I push the button on the doorbell:
- I get a doorbell alert in the UniFi Protect app
- There is a line in the homebridge log "Front Door [UVC G4 Doorbell]: Doorbell ring detected."
- I see a doorbell notification in the Home app.

Unless I'm missing something, the plugin is not getting the motion event from the Protect API. Is there anything I can look at to try and figure out why that might be happening?

One additional noteworthy detail - the motion alerts used to be working correctly in the Home app. A couple weeks ago it stopped working, I'm not sure when it was exactly.

Thanks for all your help on this and the plugin in general, I've been running it for a long time and it is very helpful.

**Comment by hjdhjd** [comment]
Alright, now we've got something to work with. What's the configuration you're using for the plugin? Did you simplify it, as I mentioned above? If not...**remove all feature options except those that relate to logging** and try again and see what works.

Good luck.

**Comment by jcg3** [comment]
Allrighty - I took out all the options so my config looks like this now:

```
        {
            "platform": "UniFi Protect",
            "controllers": [
                {
                    "address": "192.168.177.2",
                    "password": "secret",
                    "username": "secret"
                }
            ],
            "options": [
                "Enable.Log.Doorbell",
                "Enable.Log.Motion"
            ]
        },
```

Same behavior.

I went ahead and removed my other homebridge plugins (Nest, Indigo). Homebridge only has two plugins at this point - Homebridge UI and Homebridge UniFi Protect.

Same behavior. ☹️

**Comment by hjdhjd** [comment]
Are the clocks on your UCK and your Mac in sync with each other?

**Comment by jcg3** [comment]
Yes, they are in sync.

```
$ ssh unifi.local date ; date
Sat Jul  3 17:42:31 EDT 2021
Sat Jul  3 17:42:31 EDT 2021
```

**Comment by hjdhjd** [comment]
Nice touch with the command line output. :smile:

What camera types do you have installed? I presume it's just the G4 Doorbell in question? Does the Protect webUI **event log** show the motion events in question?

**Comment by jcg3** [comment]
I have a G3 Flex but it wasn't in use, so I hooked it up and motion alerts ARE working properly with it... showing in Protect, in the homebridge log, showing an alert in the Home app.

So that narrows down the problem to the Doorbell Camera specifically. The events are showing in the event page in the Protect app.

I decided to restart the Doorbell camera, and it started working. 🤦 

Thanks for all your help with trying to figure this out.
